PR: https://github.com/odoo/odoo/pull/263600

From: 044d656253f8e56fc4b6f770bad15c2d7be01f4f
From: Tanguy Quéguineur
Date: 2026-05-12 10:49:11

Breaking data model changes scores: del:7 + add:0, change matches:
-    monday_location_id = fields.Many2one("hr.work.location", related="employee_id.monday_location_id", readonly=False, string='Mondays', user_writeable=True)
-    tuesday_location_id = fields.Many2one("hr.work.location", related="employee_id.tuesday_location_id", readonly=False, string='Tuesdays', user_writeable=True)
-    wednesday_location_id = fields.Many2one("hr.work.location", related="employee_id.wednesday_location_id", readonly=False, string='Wednesdays', user_writeable=True)
-    thursday_location_id = fields.Many2one("hr.work.location", related="employee_id.thursday_location_id", readonly=False, string='Thursdays', user_writeable=True)
-    friday_location_id = fields.Many2one("hr.work.location", related="employee_id.friday_location_id", readonly=False, string='Fridays', user_writeable=True)
-    saturday_location_id = fields.Many2one("hr.work.location", related="employee_id.saturday_location_id", readonly=False, string='Saturdays', user_writeable=True)
-    sunday_location_id = fields.Many2one("hr.work.location", related="employee_id.sunday_location_id", readonly=False, string='Sundays', user_writeable=True)

Total Changes: 14

[FIX] hr: allow user to edit own location

**Steps to reproduce**
* Have a user without HR rights and linked to an employee
* With this user, open Preferences and in the calendar tab and try
to change the work location for one of the days
* Error: You do not have enough rights to access the field "version_id"
on Employee (hr.employee).

**Cause**
Issue after 72ac4b03657d617644ae75f2957aaec7acf6c1a8 which removed
SELF_READABLE_FIELDS and SELF_WRITEABLE_FIELDS.

**Change**
Use the `field_employee` function introduced in 9605045313953b4c8c734c0d52e8032e3c36bf3a
(commit message contains the explanation as to why it is necessary for
fields coming from the employee model).

opw-6127522

closes odoo/odoo#263600

X-original-commit: 953de06f5dd6d04bf28965129cefc00e23bc4973
Signed-off-by: Mathias Lebel (lebm) <lebm@odoo.com>
Signed-off-by: Tanguy Quéguineur (taqu) <taqu@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr/models/res_users.py
+++ b/addons/hr/models/res_users.py
@@ -147,13 +147,13 @@ class ResUsers(models.Model):
     is_system = fields.Boolean(compute="_compute_is_system")
     is_hr_user = fields.Boolean(compute='_compute_is_hr_user')
 
-    monday_location_id = fields.Many2one("hr.work.location", related="employee_id.monday_location_id", readonly=False, string='Mondays', user_writeable=True)
-    tuesday_location_id = fields.Many2one("hr.work.location", related="employee_id.tuesday_location_id", readonly=False, string='Tuesdays', user_writeable=True)
-    wednesday_location_id = fields.Many2one("hr.work.location", related="employee_id.wednesday_location_id", readonly=False, string='Wednesdays', user_writeable=True)
-    thursday_location_id = fields.Many2one("hr.work.location", related="employee_id.thursday_location_id", readonly=False, string='Thursdays', user_writeable=True)
-    friday_location_id = fields.Many2one("hr.work.location", related="employee_id.friday_location_id", readonly=False, string='Fridays', user_writeable=True)
-    saturday_location_id = fields.Many2one("hr.work.location", related="employee_id.saturday_location_id", readonly=False, string='Saturdays', user_writeable=True)
-    sunday_location_id = fields.Many2one("hr.work.location", related="employee_id.sunday_location_id", readonly=False, string='Sundays', user_writeable=True)
+    monday_location_id = field_employee(fields.Many2one, 'monday_location_id', comodel_name='hr.work.location', string='Mondays', user_writeable=True)
+    tuesday_location_id = field_employee(fields.Many2one, 'tuesday_location_id', comodel_name='hr.work.location', string='Tuesdays', user_writeable=True)
+    wednesday_location_id = field_employee(fields.Many2one, 'wednesday_location_id', comodel_name='hr.work.location', string='Wednesdays', user_writeable=True)
+    thursday_location_id = field_employee(fields.Many2one, 'thursday_location_id', comodel_name='hr.work.location', string='Thursdays', user_writeable=True)
+    friday_location_id = field_employee(fields.Many2one, 'friday_location_id', comodel_name='hr.work.location', string='Fridays', user_writeable=True)
+    saturday_location_id = field_employee(fields.Many2one, 'saturday_location_id', comodel_name='hr.work.location', string='Saturdays', user_writeable=True)
+    sunday_location_id = field_employee(fields.Many2one, 'sunday_location_id', comodel_name='hr.work.location', string='Sundays', user_writeable=True)
 
     @api.depends_context('uid')
     def _compute_is_system(self):
